frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.html

```diff
@@ -43,10 +48,21 @@ <h2 *ngIf="existing"></h2>
         <div class="field-separator"></div>
         <mat-form-field appearance="fill">
           <mat-label>Port</mat-label>
-          <input matInput inputmode="numeric" formControlName="port" />
-          <mat-error *ngIf="form.controls.port.errors"
-            >The port is invalid.</mat-error
+          <input
+            matInput
+            inputmode="numeric"
+            type="number"
```

Comment:
It should be a numeric string but not a number. The number type should only be used for mathematical numbers, not strings that appear to be composed of digits. The `inputmode="numeric"` is here to hint that it’s composed of digits. https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode

Another issue is that the number type allows floats (and Angular then truncates then to integers, but still).

A blog post on the matter: https://stackoverflow.blog/2022/09/15/why-the-number-input-is-the-worst-input/

Reply:
Interesting, thanks for sharing the articles. Actually, I always thought that a port is a number - but even this is not the case: https://stackoverflow.com/a/47992797. But yeah, this is a pretty rare case - I think we can stick with numbers, but I'll remove the `type="number"` annotation.